docs/general/developer-resources/emerald-paratime/writing-dapps-on-emerald.md

```diff
@@ -4,39 +4,173 @@ description: How to write and deploy a dApp to Oasis Emerald
+underlying algorithm for signing the transactions is [ed25519] on the Consensus
+layer and both [ed25519] and [ECDSA] in Emerald. The ed25519 scheme is used
+mostly by the Emerald compute nodes for managing their computation rewards. For
+signing your dApp-related transactions on Emerald you will probably want to use
+ECDSA since this is the de facto scheme supported by Ethereum wallets and
+libraries.
+
+Finally, the ParaTimes are not allowed to directly access your tokens stored in
+Consensus layer addresses. You will need to **deposit** tokens from your consensus
+account to Emerald. Consult the [How to transfer ROSE into Emerald ParaTime]
+[how-to-deposit-rose] chapter to learn more.
+
+
+[overview]: ../../oasis-network/overview.md
+[ed25519]: https://en.wikipedia.org/wiki/EdDSA#Ed25519
+[ECDSA]: https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm
+[how-to-deposit-rose]: ../../manage-tokens/how-to-transfer-rose-into-emerald-paratime.mdx
+[Testnet faucet]: https://faucet.testnet.oasis.dev/
 
 ## Testnet and Mainnet
 
-The Oasis network currently has, similar to some other blockchains, two major public deployments: the [Mainnet](../../oasis-network/network-parameters.md) and the [Testnet](../../foundation/testnet/README.md). The native tokens for the networks are called ROSE and TEST respectively. Each deployment has its own state, a different set of validators and ParaTimes. The state of the Mainnet is considered immutable for indefinite time, while the data on the Testnet can be subject to wipe in the future.
+The Oasis network currently has, similar to some other blockchains, two major
+public deployments: the [Mainnet][network-parameters] and the [Testnet]. The
+native tokens are called ROSE and TEST respectively. Each deployment has its own
+state, a different set of validators and ParaTimes. The state of the Mainnet is
+considered immutable for indefinite time, while the data on the Testnet can be
+subject to wipe in the future.
 
-Emerald is deployed similarly: the [Emerald Mainnet](./README.mdx#mainnet) is deployed on the Oasis Mainnet network, and the [Emerald Testnet](./README.mdx#testnet) on the Oasis Testnet network. The Emerald state on the Mainnet is stable. Testnet, apart from running the unstable version of the code and being prone to bugs, can have the state deliberately wiped either on the Emerald ParaTime layer or on the Oasis Testnet network level, so you should **never deploy a production service on the Testnet**!
+The Emerald ParaTime is deployed similarly: the [Emerald Mainnet] is deployed on the Oasis Mainnet
+network while the [Emerald Testnet] on the Oasis Testnet network. The Emerald state on the Mainnet
+is stable. Testnet, apart from running the unstable version of the code and being prone to bugs, can
+have the state deliberately wiped either on the Emerald ParaTime layer or on the Oasis Testnet
+network level, so you should **never deploy a production service on the Testnet**!
```

Comment:
Move the "you should **never deploy a production service on the Testnet**!" part to a danger tool tip.